src/Kentico.Xperience.UMT/InfoAdapter/ContentItemSimplifiedAdapter.cs

```
@@ -173,6 +175,39 @@ public ContentItemInfo Adapt(IUmtModel input)
             var itemDataInfo = (ContentItemDataInfo)itemDataAdapter.Adapt(itemDataModel);
             itemDataAdapter.ProviderProxy.Save(itemDataInfo, itemDataModel);
 
+            var contentItemReferenceFormFieldInfos = fi.GetFields(true, true).Where(x => x.DataType == contentItemReferenceDataTypeName);
+
+            if (customData is not null && contentItemReferenceFormFieldInfos is not null)
+            {
+                var contentItemReferenceModels =
+                    CreateContentItemReferenceModels(contentItemReferenceFormFieldInfos, customData, commonDataModel);
+
+                if (contentItemReferenceModels is not null)
+                {
+                    foreach (var referenceModel in contentItemReferenceModels)
+                    {
+                        var targetItem = Provider<ContentItemInfo>.Instance.Get()
+                            .WhereEquals(nameof(ContentItemInfo.ContentItemGUID), referenceModel.ContentItemReferenceTargetItemGuid)
+                            .FirstOrDefault();
+
+                        ArgumentNullException.ThrowIfNull(targetItem);
```

Please throw an exception with a message from which the user can unambiguously understand what went wrong, for which content item, linked content item, ...